lightning/src/ln/channelmanager.rs

```diff
@@ -1551,10 +1553,28 @@ impl ChannelDetails {
 			inbound_htlc_minimum_msat: Some(context.get_holder_htlc_minimum_msat()),
 			inbound_htlc_maximum_msat: context.get_holder_htlc_maximum_msat(),
 			config: Some(context.config()),
+			channel_shutdown_state: context.shutdown_state(),
 		}
 	}
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+/// Further information on the details of the channel shutdown
+pub enum ChannelShutdownState {
+	/// Channel has not sent or recieved a shutdown message
+	NotShuttingDown,
+	/// Local node has sent a shutdown message for this channel
```

Comment:
Looks like this will also be the state if the remote sent a shutdown message, so we either need separate variants for local vs remote initiated shutdown or a docs update

Reply:
during my tracing of the code i found that if a remote sent a shutdown, local node goes directly into `ResolvingHTLCs` or `ClosingSignNegotiation`

do take a look at `lightning/src/ln/shutdown_tests.rs` line 139 in this PR.